### ISSUE TITLE
Use sqlite for stats plugin

### ISSUE THREAD

**Issue by malishev** [yagop/telegram-bot#61]
**Problem:** my bot is used very intensively. So at one moment the stats.lua file becomes so big that it cannot be written into anymore. Because of some reason it becomes empty and the following error occurs:

![2015-02-07 23 31 58](https://cloud.githubusercontent.com/assets/2178368/6093822/88ebb592-af21-11e4-9e00-b25672f20ae8.png)

So i suggest using sqlite for storing and displaying stats.

**Comment by yagop** [comment]
Hi, ¿How big is your stats file? Serialization is done with https://github.com/pkulchenko/serpent is a fast lib but if the file is too big I don't know how it will behave. That error tells that '_stats' is outside the scope http://stackoverflow.com/questions/22346444/attempt-to-index-upvalue-which-is-a-nil-value so it can be indexed.

I allays try to maintain the bot core as simple as possible, including SQLite might be a big mess.

**Comment by malishev** [comment]
Well, i'm not sure if it crashes because of the size, but the fact is - it crashes. I know you've already know about intensivity:
![image](https://cloud.githubusercontent.com/assets/2178368/6097066/8c662b50-afbf-11e4-8e32-2fbee50e9fbf.jpg)

Imagine 100 more of such a groups :smiley:

**Comment by malishev** [comment]
Maybe blocking the stats writing for groups is a solution?

**Comment by yagop** [comment]
I use stats for groups, another solution might be save stats only once per minute with cron feature.

**Comment by yagop** [comment]
Take a look of the `stats.lua` file, it doesn’t save the messages, just only the number per user. Despite of the huge number of messages sent to a group it will only increment a value. If the users send too fast it might block, but is very uncommon.

**Comment by malishev** [comment]
Yeah, i guess you're right. Using database for just incrementing the numbers is superfluous and unnecessary.

So i think maybe a blacklist with the IDs of groups or users in, for example, `config.lua` file will be the compromise solution? It will not solve the problem with the stats plugin, but may help to prevent its crashing.

**Comment by yagop** [comment]
I know that File as DB sucks.

Possible fixes:
- Disable stats and other plugins that uses lua files as db.
- Use mysql / pg / others reals DB (better one)

**Comment by yagop** [comment]
Since REDIS and Antiflood this should be fixed
